packages/api/src/submittable/createClass.ts

```diff
@@ -349,6 +377,32 @@ export function createClass <ApiType extends ApiTypes> ({ api, apiType, blockHas
         }
       }
     };
+
+    /**
+     * When a signer includes `signedTransaction` within the SignerResult this will validate
+     * specific fields within the signed extrinsic against the original payload that was passed
+     * to the signer.
+     */
+    #validateSignedTransaction = (signerPayload: SignerPayload, signedExt: Extrinsic): void => {
```

Comment:
Note that this validation logic only covers the current case of Polkadot / Kusama relay chains. Other chains (and future relay chain upgrades) may have other signed extensions which would be skipped if we go for this function (for example, the asset ID for fee payment in the Asset Hubs).

Therefore my suggestion is either to:
1. Do no validation at all before submission, and clearly document the behaviour of the`signedTransaction` mode,
2. Only validate the payload method, also documenting the implications, or
3. do a byte-level comparison of the `included_in_extrinsic` kind of signed extensions given by the relevant chain, this is, the "extra-data" present in the `signedTransaction`.

My suggestion is to go for 1 or 2 as discussed beforehand.

Reply:
I agree with what @IkerAlus proposes, however as discussed, I'm more of a proponent of option (2): I would validate that `payload.method` remains the same.

Even though adding this check does not inherently improves the security of the system as the signer can broadcast any transaction it wants, I believe it does help to take a more conservative approach until the ecosystem can move to a bigger, different structure.

Reply:
Sounds good, I'll just ensure that the method is the same and that's it!